inbox/mailsync/backends/imap/common.py

```diff
@@ -230,69 +253,109 @@ def remove_deleted_uids(account_id, folder_id, uids):
     """
     if not uids:
         return
-    deleted_uid_count = 0
-    for uid in uids:
-        # We do this one-uid-at-a-time because issuing many deletes within a
-        # single database transaction is problematic. But loading many
-        # objects into a session and then frequently calling commit() is also
-        # bad, because expiring objects and checking for revisions is O(number
-        # of objects in session), resulting in quadratic runtimes.
-        # Performance could perhaps be additionally improved by choosing a
-        # sane balance, e.g., operating on 10 or 100 uids or something at once.
```

Comment:
This comment says it all, it was here from the very start. We no longer load as many objects to session as we limited that in previous PR.